### PR TITLE
Elisp updates

### PR DESCRIPTION
**Pull request by bhankas**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bhankas** [comment]
@AndersonTorres @adisbladis I think I've followed most steps here, but despite multiple attempts `update` script does not finish successfully.

There are a bunch of new grammars in `tree-sitter-langs`, including at least one that I'm interested in (`gleam-lang`), hence the whole thing.

**Comment by AndersonTorres** [comment]
Manual packages are not easy to update. But don' t worry about them - updating from overlay should be sufficient.

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/tree-sitter-langs/update-defaults.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def build_attr(nixpkgs, attr: str) -> str:
 
 if __name__ == "__main__":
     cwd = dirname(abspath(__file__))
-    nixpkgs = abspath(join(cwd, "../../../../../.."))
+    nixpkgs = abspath(join(cwd, "../../../../../../.."))
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/manual-packages/tree-sitter-langs/update-defaults.py:58]: This path should point to which file?
To the root default.nix?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Remove this "Merge branch 'NixOS:master' into master".

In pull requests we rebase instead.

File: pkgs/applications/editors/emacs/elisp-packages/manual-packages/tree-sitter-langs/update-defaults.py (hunk 1 above)
Comment (bhankas) [reply pkgs/applications/editors/emacs/elisp-packages/manual-packages/tree-sitter-langs/update-defaults.py:58]: It should point to root of nixpkgs repo. Unsure on the original reasoning, but without this change, this script fails to execute.

**Comment by AndersonTorres** [comment]
I suggest you to update the grammars in a separated pull request.

**Comment by bhankas** [comment]
> I suggest you to update the grammars in a separated pull request.

Done! Kept all the elisp changes here though, since this is after all, about Elisp packages.

**Comment by AndersonTorres** [comment]
Not dropped yet.

**Comment by bhankas** [comment]
Oh you meant all tree-sitter changes, I misunderstood. Fixed now.
Anderson Torres ***@***.***> writes:

> Not dropped yet.


--

**Comment by bhankas** [comment]
Raised #268477 for tree-sitter changes.
--

**Comment by samueltardieu** [comment]
Shouldn't those two PRs be submitted against staging?

**Comment by AndersonTorres** [comment]
> Shouldn't those two PRs be submitted against staging?

Unneeded.

In the particular case of Emacs Lisp packages, 

1. the "mass rebuild" flags appear just because they are too many
2. they are just a bunch of small files, they have negligible building time
3. they are mostly leaf packages, no other packages depend on them
4. therefore they do not slow down the unstable channels

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**